### PR TITLE
[DOCS] Add brew install instructions. Closes #42914

### LINKED ISSUE #42914: [DOCS] Add Homebrew installation instructions
Update the installation info to:

- Add brew as an installation option on the main [install page](https://www.elastic.co/guide/en/elasticsearch/reference/current/install-elasticsearch.html)
- Add a topic called "Install Elasticsearch on macOS with Homebrew" for the brew instructions and give it the anchor [[brew]]

To use brew to install the stack components, users will first need to “tap” the Elastic Homebrew repo.

### PR DESCRIPTION
**Pull request by debadair**

### CODE REVIEW

File: docs/reference/setup/install/brew.asciidoc [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+install {es}:
+
+[source,sh]
+-------------------------
+brew install elastic/tap/elasticsearch
+-------------------------
+
+This installs the most recently released default distribution of {es}.
+Append the version number to install a previous version, for example
+`elastic/tap/elasticsearch@6.8`. To install the OSS distribution,
```
Comment (dedemorton) [inline docs/reference/setup/install/brew.asciidoc]: Should this just say `brew install elasticsearch@6.8`. I apologize for my ignorance (I don't use brew often), but the command you show results in:

Error: No available formula with the name "elastic/tap/elasticsearch@6.8"

Comment (debadair) [reply docs/reference/setup/install/brew.asciidoc]: You have to specify the elastic tap. The example is bogus, because the 6.8 version isn't available.

**Comment by debadair** [comment]
Removed the specific version example per @jasontedor and changed install instructions to use `elasticsearch-full`.

**Review by jasontedor — APPROVED** [review]
LGTM.

**Pull request merged**